jbmc/src/java_bytecode/java_bytecode_parser.cpp

```diff
@@ -1640,6 +1640,48 @@ void java_bytecode_parsert::rclass_attribute(classt &parsed_class)
     parsed_class.attribute_bootstrapmethods_read = true;
     read_bootstrapmethods_entry(parsed_class);
   }
+  else if(attribute_name == "InnerClasses")
+  {
+    u2 number_of_classes = read_u2();
+    u4 number_of_bytes_to_be_read = number_of_classes * 8 + 2;
+    INVARIANT(
+      number_of_bytes_to_be_read == attribute_length,
```

Comment:
I see what you are saying, but I am not sure that it's best. If the bytes don't match, I don't think we can guarantee that anything makes sense because that would mean that the bytecode spec has changed basically, and so should probably just fail hard.